packages/core/src/indexing/index.ts

```diff
@@ -317,6 +296,10 @@ export const createIndexing = ({
       context.db = db;
       for (let i = 0; i < events.length; i++) {
         const event = events[i]!;
+
+        client.event = event;
+        context.client = client.getClient(networkByChainId[event.chainId]!);
```

Comment:
I think we're indeed re-running a lot of client initialization code here - might get optimized but worth fixing IMO